File: test_archives/cpp_tests/batch5/test_function9.cpp
#include "../catch.hpp"
#include "../functions/function9.cpp"

TEST_CASE( "Testing the process_data() function", "[process_data]" ) {
    SECTION("Testing process_data() with different inputs") {
        REQUIRE(function9("Test", 1, {1, 2, 3}).process_data() == 0);
        REQUIRE(function9("Test2", 2, {3, 4, 5}).process_data() == 0);
    }
}

File: test_archives/py_tests/batch5/test_function9.py
from functions.function9 import function9
import pytest

@pytest.mark.parametrize("string_input, int_input, list_input, expected_result", 
[
    ("Test", 1, [1, 2, 3], 0),
    ("Test2", 2, [3, 4, 5], 0)
])
def test_process_data(string_input, int_input, list_input, expected_result):
    instance = function9(string_input, int_input, list_input)
    assert instance.process_data() == expected_result